pkg/controller/cheinstallation/cheinstallation_controller.go

```diff
@@ -101,67 +102,59 @@ func (r *ReconcileCheInstallation) Reconcile(request reconcile.Request) (reconci
 		return reconcile.Result{}, err
 	}
 
-	err = r.ensureCheInstallation(reqLogger, cheInstallation)
-	return reconcile.Result{}, err
-}
-
-func (r *ReconcileCheInstallation) ensureCheInstallation(logger logr.Logger, cheInstallation *v1alpha1.CheInstallation) error {
```

Comment:
IMO, `ensureCheInstallation()` was a good abstraction to have `Reconcile()` small and easy to follow.  We can have `ensureCheInstallation()` returns requeue=bool and in `Reconcile()` we actually deal with `reconcile.Request` .. something what done [here](https://github.com/nurali-techie/toolchain-operator/pull/1/files#diff-b09f4db8198fe6aeba0154bd7f8fdfbeR118).  This is simple where we are just passing bool from `ensureCheInstallation()` to `Reconcile()` and it's enough for all the cases.

Reply:
code refactored in 0ab8b1f